ldes-server-infra-postgres/postgres-pagination-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/pagination/postgres/batch/BucketPartitioner.java

```diff
@@ -1,35 +1,47 @@
 package be.vlaanderen.informatievlaanderen.ldes.server.pagination.postgres.batch;
 
+import org.springframework.batch.core.configuration.annotation.StepScope;
 import org.springframework.batch.core.partition.support.Partitioner;
 import org.springframework.batch.item.ExecutionContext;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
 
 import java.util.Map;
 import java.util.stream.Collectors;
 
 @Component
-public class BucketPartitioner implements Partitioner {
+@StepScope
+public class BucketPartitioner implements Partitioner{
 	static final String SQL = """
-			SELECT bucket_id
-			FROM page_members
-			WHERE page_id IS NULL
-			GROUP BY bucket_id
-			ORDER BY COUNT (member_id) DESC
-			LIMIT ?
+			SELECT pm.bucket_id
+			FROM page_members pm
+			JOIN buckets b on b.bucket_id = pm.bucket_id
+			JOIN views v on v.view_id = b.view_id
+			JOIN collections c on c.collection_id = v.collection_id
+			WHERE c.name = ? AND v.name = ? AND page_id IS NULL
+			GROUP BY pm.bucket_id
 			""";
 
 	private final JdbcTemplate jdbcTemplate;
+	private final Map<String, Object> jobParameters;
 
-	public BucketPartitioner(JdbcTemplate jdbcTemplate) {
+
+	public BucketPartitioner(JdbcTemplate jdbcTemplate, @Value("#{jobParameters}") Map<String, Object> jobParameters) {
 		this.jdbcTemplate = jdbcTemplate;
+		this.jobParameters = jobParameters;
 	}
 
 	@Override
+	@Transactional
 	public Map<String, ExecutionContext> partition(int gridSize) {
-		return jdbcTemplate.queryForList(SQL, Long.class, gridSize).stream()
+		String collectionName = (String) jobParameters.get("collectionName");
+		String viewName = (String) jobParameters.get("viewName");
+
+		return jdbcTemplate.queryForList(SQL, Long.class, collectionName, viewName).stream()
```

Comment:
It is.. we never set the value itself explicitly, so we can do without. Otherwise we got strange behaviour that fragmentation of a view not always processes all buckets